compiler/rustc_error_codes/src/error_codes/E0472.md

```diff
@@ -0,0 +1,31 @@
+Inline assembly (`asm!`) is not supported on this target.
+
+Example of erroneous code:
+
+```ignore (cannot-test-because-unable-to-change-target)
+// compile-flags: --target sparc64-unknown-linux-gnu
+#![no_std]
+
+use core::arch::asm;
+
+fn main() {
+    unsafe {
+        asm!(""); // error: inline assembly is not supported on this target
+    }
+}
+```
+
+The Rust compiler does not support inline assembly, with the `asm!` macro
+(previously `llvm_asm!`), for all targets. All Tier 1 targets do support this
+macro but support among Tier 2 and 3 targets is not guaranteed (even when they
+have `std` support). Note that this error is related to
+`error[E0658]: inline assembly is not stable yet on this architecture`, but
```

Comment:
This could link to the page for E0658

Reply:
My mistake, I was going to do that. On second thought I think that it is important to mention that E0658 is generic for all errors where a unstable feature needs to be enabled; E0658 doesn't provide additional context for `inline assembly is not stable yet on this architecture`. I'm still happy to add a link if you want though.

Reply:
I see. I'm okay either way.